web-client/src/view/BloodCenters/CenterDetails/GoogleMap.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef } from 'react';
+
+const addSingleMarkers = ({ locations, map }: {
+    locations: ReadonlyArray<google.maps.LatLngLiteral>;
+    map: google.maps.Map | null | undefined;
+}) =>
+    locations.map(
+        position =>
+            new google.maps.Marker({
+                position,
+                map,
+            }),
+    );
+
+const GoogleMap = ({ center, zoom }: {
+    center: google.maps.LatLngLiteral;
+    zoom: number;
+}) => {
+    const ref = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (ref.current) {
+            const map = new window.google.maps.Map(ref.current, {
```

Comment:
i think this logic is everytime the zoom level or center changes reinitializing the map, right?

Reply:
Yes, I used this article - https://medium.com/inato/easy-google-maps-setup-using-google-maps-api-in-react-51e0fef5aefe